commercetools-convenience/src/test/java/io/sphere/sdk/meta/CategoryDocumentationIntegrationTest.java

```diff
@@ -299,15 +381,25 @@ private static void withProductInCategory(final BlockingSphereClient client, fin
     }
 
     private static CategoryTree fetchCurrentTree() {
-        final CompletionStage<List<Category>> categoriesStage = QueryExecutionUtils.queryAll(client(), CategoryQuery.of());
-        final List<Category> categories = SphereClientUtils.blockingWait(categoriesStage, Duration.ofMinutes(5));
+        final CompletionStage<List<List<Category>>> categoryPagesStage = QueryExecutionUtils.queryAll(client(), CategoryQuery.of(), (categories -> categories));
+        final List<List<Category>> categoryPages = SphereClientUtils
+            .blockingWait(categoryPagesStage, Duration.ofMinutes(5));
+        final List<Category> categories = categoryPages.stream()
+                                                       .flatMap(List::stream)
+                                                       .collect(toList());
         return CategoryTree.of(categories);
     }
 
     private CategoryTree createCategoryTree() {
         //stuff from previous example
-        final CompletionStage<List<Category>> categoriesStage = QueryExecutionUtils.queryAll(client(), CategoryQuery.of());
```

Comment:
When you changed the code to also provide the old API, you should be able to revert these changes.

Reply:
reverted legacy API: 16d5ea6